kentik_api_library/kentik_api/public/site.py

```diff
@@ -20,7 +20,8 @@ def __init__(
         self._company_id = company_id
 
     @property
-    def id(self) -> Optional[int]:
+    def id(self) -> int:
+        assert self._id is not None
```

Comment:
Returning None when the ID (or any other attribute) is not yet set is ok. I don't think we should raise an exception here. If you just want to make mypy happy, I think Optional[int] as a return type should do the trick?